copy_generated_types_from_lemmy_js_client.sh

```diff
@@ -45,15 +54,44 @@ for filename in *.ts;
   sed -i 's/registration_mode: String/registration_mode: RegistrationMode/g' "$kt_file"
   sed -i 's/feature_type: String/feature_type: PostFeatureType/g' "$kt_file"
   
-  # Add = null to any lines containing ?
-  sed -i '/\?/ s/$/ = null/' "$kt_file"
+  # Add = null to any lines containing ? if it is not a typealias
+  if ! grep -q "typealias" "$kt_file"; then
+      sed -i '/\?/ s/$/ = null/' "$kt_file"
+  fi
 
   # Change these interfaces to data classes
   sed -i 's/interface /data class /g' "$kt_file"
   sed -i 's/ {/(/g' "$kt_file"
   sed -i 's/}/)/g' "$kt_file"
   sed -i '/:/ s/$/,/' "$kt_file"
 
+  # Add @Parcelize annotation
+  sed -i 's/data class/import android.os.Parcelable\nimport kotlinx.parcelize.Parcelize\n\n@Parcelize\ndata class/g' "$kt_file"
+
+  # Add Parcelable inheritance
+  if grep -q "data class" "$kt_file"; then
+      echo " : Parcelable" >> "$kt_file"
+  fi
+
+  # Adds @Immutable annotation to classes containing lists, to mark them stable
+  if grep -q "List<" "$kt_file"; then
```

Comment:
Since you're regenerating all of these anyway, now would probabably be a good time to convert all the `List` to `ImmutableList`, `L41` above.

Reply:
I tried that approach but refrofit/whatever makes the api would not work with it. Was easier to mark it stable.

Reply:
Gotcha, nbd.